src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -120,19 +120,23 @@ public static partial class JsonSerializer
         }
 
         internal static void CreateDataExtensionProperty(
```

Comment:
The reflection being added to this method will occur during deserialization (as opposed to during the serializer's warm-up, when `JsonClassInfo`s and `JsonPropertyInfo`s are created), and will lead to bad throughput performance.

I'll come back to this in a follow-up review, but I think the type metadata being retrieved here can be cached on the `JsonClassInfo` for the dictionary type during warm-up.